internal/pkg/service/common/etcdop/op/atomic_test.go

```diff
@@ -13,10 +14,290 @@ import (
 	etcd "go.etcd.io/etcd/client/v3"
 
 	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/iterator"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/op"
 	"github.com/keboola/keboola-as-code/internal/pkg/utils/etcdhelper"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/etcdlogger"
 )
 
+type atomicOpTestCase struct {
+	Name               string
+	Prepare            func(t *testing.T, client etcd.KV) []op.Op
+	ReadPhase          func(t *testing.T, client etcd.KV) []op.Op
+	BreakingChange     func(t *testing.T, client etcd.KV) []op.Op
+	ExpectedWritePhase string
+}
+
+func TestAtomicOp(t *testing.T) {
+	t.Parallel()
+
```

Comment:
In the first commit a added better tests for the AtomicOp.

Each case it tested twice:
- First run tests the AtomicOp without the `BreakingChange` between Read and Write phase.
- Second run tests it with the `BreakingChange`.